editor/src/core/shared/deep-equality-instances.ts

```diff
@@ -0,0 +1,188 @@
+import { HigherOrderControl } from '../../components/canvas/canvas-types'
+import {
+  DerivedState,
+  transientCanvasState,
+  TransientCanvasState,
+  TransientFileState,
+} from '../../components/editor/store/editor-state'
+import {
+  arrayDeepEquality,
+  createCallFromEqualsFunction,
+  createCallFromIntrospectiveKeepDeep,
+  KeepDeepEqualityCall,
+  keepDeepEqualityResult,
+  KeepDeepEqualityResult,
+} from './deep-equality'
+import { PropertyPath, TemplatePath } from './project-file-types'
+import * as TP from './template-path'
+import * as PP from './property-path'
+import { JSXElementName } from './element-template'
+
+export const TemplatePathKeepDeepEquality: KeepDeepEqualityCall<TemplatePath> = createCallFromEqualsFunction(
+  (oldPath: TemplatePath, newPath: TemplatePath) => {
+    return TP.pathsEqual(oldPath, newPath)
+  },
+)
+
+export const PropertyPathKeepDeepEquality: KeepDeepEqualityCall<PropertyPath> = createCallFromEqualsFunction(
+  (oldPath: PropertyPath, newPath: PropertyPath) => {
+    return PP.pathsEqual(oldPath, newPath)
+  },
+)
+
+export function JSXElementNameKeepDeepEqualityCall(
+  oldName: JSXElementName,
+  newName: JSXElementName,
+): KeepDeepEqualityResult<JSXElementName> {
+  let areEqual: boolean = true
+
+  let baseVariable: string
+  if (oldName.baseVariable === newName.baseVariable) {
+    baseVariable = oldName.baseVariable
+  } else {
+    areEqual = false
+    baseVariable = newName.baseVariable
+  }
+
+  const propertyPathResult = PropertyPathKeepDeepEquality(
+    oldName.propertyPath,
+    newName.propertyPath,
+  )
+  const propertyPath = propertyPathResult.value
+  areEqual = areEqual && propertyPathResult.areEqual
+
+  if (areEqual) {
+    return keepDeepEqualityResult(oldName, true)
+  } else {
+    const elementName: JSXElementName = {
+      baseVariable: baseVariable,
+      propertyPath: propertyPath,
+    }
+    return keepDeepEqualityResult(elementName, areEqual)
```

Comment:
Might as well replace `areEqual` with `false` here for clarity

Reply:
Sure.

Reply:
Made this changes in f1b06eb0.